### ISSUE TITLE
i can't find where you use finetuning on first frame in your code-pytorch

### ISSUE THREAD

**Issue by zouying-sjtu** [kmaninis/OSVOS-PyTorch#30]

**Comment by kmaninis** [comment]
Hi, online training on the first frame happens in [this](https://github.com/kmaninis/OSVOS-PyTorch/blob/36562efeed51dd994b44329ea3cff5e0d76e83ac/train_online.py#L96) script.

The data loader uses only the first frame in [this](https://github.com/kmaninis/OSVOS-PyTorch/blob/36562efeed51dd994b44329ea3cff5e0d76e83ac/dataloaders/davis_2016.py#L59) part of the dataloader.

I hope it helps.

**Comment by zouying-sjtu** [comment]
one more question, besides the dataset is different, is there any other difference between training and finetuning. thanks

**Comment by kmaninis** [comment]
Could you be more specific?

**Comment by zouying-sjtu** [comment]
when you train the model, you use many data-label pairs from trainset. when you finetune, you just use the pair of the first frame. is there other more manipulation you do?

**Comment by kmaninis** [comment]
The offline and online training processes are almost identical. Some details are different, eg. the [deeply supervised loss](https://github.com/kmaninis/OSVOS-PyTorch/blob/36562efeed51dd994b44329ea3cff5e0d76e83ac/train_parent.py#L144) in offline training, and parameters such as nAveGrad, since more speed is needed during online training.
